### PR TITLE
Refactor load_taxonomic_resources to increase speed

### PR DESCRIPTION
**Pull request by dfalster**
This PR refactors a few functions to increase speed. 

The time to run `load_taxonomic_resources` has dropped from 15.0s to 2.2s (on Daniel's MacBook Pro M2)

### CODE REVIEW

**Comment by dfalster** [comment]
for #187 

@ehwenk you can add your tweaks for name processing after this branch has been merged,

**Comment by wcornwell** [comment]
```
could not find function "strip_names_2"
```
i think it's in an example somewhere?

**Comment by dfalster** [comment]
Yes, I'l fix that now

File: R/standardise_names.R [hunk 1]
```diff
@@ -107,20 +107,50 @@ standardise_names <- function(taxon_names) {
...
   
-  genus <- str_split_i(taxon_name, " ", 1)
-  word2 <- str_split_i(taxon_name, " ", 2)
+  genus <- str_split_i(taxon_name, " ", 1) %>% stringr::str_to_sentence()
   
   # Deal with names that being with x, 
   # e.g."x Taurodium x toveyanum" or "x Glossadenia tutelata"
-  i <- stringr::str_to_lower(genus) == "x"
-  genus[i] <- paste("x", word2[i])
+  i <- !is.na(genus) & genus =="X"
```
Comment (ehwenk) [inline R/standardise_names.R:120]: Won't this just look for uppercase "X"? It needs to consider either "x" or "X"

File: vignettes/articles/function_notes.Rmd [hunk 2]
```diff
@@ -218,9 +218,9 @@ taxon_names            #input vector of taxon names
 
 **output**: A character vector of stripped taxonomic names, with subtaxa designations, special characters, and extra whitespace removed, and all letters converted to lowercase.
 
-### strip_names_2
+### strip_names_extra
 
-**description**: Given a vector of taxonomic names, this function removes subtaxa designations ("subsp.", "var.", "f.", and "ser"), additional filler words and characters (" x " [hybrid taxa], "sp."), special characters (e.g., "-", ".", "(", ")", "?"), and extra whitespace. The resulting vector of names is also converted to lowercase.
+**description**: Sugegested to run after strip_names, given a vector of taxonomic names, this function removes additional filler words and characters (" x " [hybrid taxa], "sp."). The resulting vector of names is also converted to lowercase.
```
Comment (ehwenk) [inline vignettes/articles/function_notes.Rmd:223]: Fix typo - Suggested

**Review by ehwenk — APPROVED** [review]
Line 120 in standardise_names needs to be changed (I think) to match both uppercase and lowercase X and one other spelling mistake, but otherwise good.

**Pull request merged**